### PR TITLE
Add LDAP search example

### PR DESCRIPTION
**Pull request by wfernandes**
Adds example describing how to do a more complex LDAP search query over TLS connection.

### CODE REVIEW

**Review by patrick-strata — COMMENTED** [review]
@wfernandes I tried to commit some changes to the PR for easy feedback but do not have write permissions to it.

Proposed README updates:

```
# LDAP Search Service Extension

There are times when you may find the need to construct a unique LDAP search 
query that isn't easily available via the LDAP connector. For example,
returning multiple result entries.

This example will show you how to query LDAP securely by upgrading the TCP 
connection to TLS and then make a request to the LDAP server to retrieve groups 
with a specific `uniqueMember` attribute. 

The structure of the LDAP can be seen in the [`example.ldif`](./example.ldif) file.

For more information regarding the `maverics/ldap` pkg, please refer to
the [Service Extension Maverics package documentation][maverics-ldap-docs].

## Setup

Please reference the [maverics.yaml](maverics.yaml) configuration file and the
Service Extension files it references for a set action items specified with`TODO`. 
These action items are changes necessary to get this example running.

## Testing

1. Complete all the action items specified by `TODO`s
1. Restart the Orchestrator, and ensure it starts successfully.
1. Navigate to the URL the Orchestrator is listening on in your browser: 
   e.g. https://localhost/headers.
1. You should now be redirected your specified IDP and prompted for authentication.
1. After successfully logging in, the custom headers that were created will be sent
   to the upstream application. This can be confirmed by verifying the
   `EXAMPLE-GROUPS` header is rendered on the `/headers` page of the sample app.

[maverics-ldap-docs]: https://docs.strata.io/orchestrator-reference/service-extensions/maverics-packages#package-maverics-ldap
```

File: ldap-search/example.ldif [hunk 1]
```diff
@@ -0,0 +1,75 @@
+version: 1
+
+dn: dc=sonarsystems,dc=com
```
Comment (patrick-strata) [inline ldap-search/example.ldif:3]: Should we sanitize the sonarsystems to something else? 
I recall there was a desire to standardize onto something for our examples but I do not recall if a decision was made.

File: ldap-search/maverics.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+connectors:
+  # The 'name' property is used as a unique ID that the Service Extensions depend on.
+  # Please ensure the Service Extensions are updated if the name of the connectors change.
+  # Additionally, please note that the 'oauthRedirectURL' used in the two connectors
+  # must be unique.
+  - name: azure
+    type: azure
+    authType: saml
+    samlMetadataURL: { SAML METADATA URL }
```
Comment (patrick-strata) [inline ldap-search/maverics.yaml:54]:
```suggestion
    samlMetadataURL: { SAML METADATA URL } # TODO: Configure your IDP.
```

File: ldap-search/secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+secrets:
+  serviceAccountUsername: cn=service,dc=sonarsystems,dc=com
+  serviceAccountPassword: uCgfpeeyGk9Vgf$*$
+  ldapCACert: |+
+    -----BEGIN CERTIFICATE-----
+    <CERT>
```
Comment (patrick-strata) [inline ldap-search/secrets.yaml:6]:
```suggestion
    <CERT> # TODO: Add your LDAP CA Cert.
```

**Review by patrick-strata — COMMENTED** [review]
This is looking good @wfernandes I just added some comments to help the user find all the areas they need to change by being able to search on `TODO:`.


I also proposed some language updates to the readme based on the doc guidelines: https://maverics.atlassian.net/wiki/spaces/DOC/pages/17072135/Strata+Style+Guide#Brand-voice-%26-tone

File: ldap-search/example.ldif (hunk 1 above)
Comment (wfernandes) [reply ldap-search/example.ldif:3]: Yeah, I thought about that too. I'm happy to change it to `examples` to be more generic.

**Review by patrick-strata — APPROVED** [review]
Nice job!

**Pull request merged**